### PR TITLE
Add min SDK file

### LINKED ISSUE #172: Tap Jacking checks written and tested

### PR DESCRIPTION
**Pull request by nwalsh1995**
Resolves #172 

This plugin should be for any checks that make a decision based of the min sdk, like the tapjacking plugin does.

### CODE REVIEW

File: qark/plugins/manifest/min_sdk.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+               "https://media.blackhat.com/ad-12/Niemietz/bh-ad-12-androidmarcus_niemietz-WP.pdf")
+
+
+class MinSDK(BasePlugin):
+    """This plugin will create issues depending on the manifest's minimum SDK. For instance
+    tapjacking is only protected when min_sdk > 9 (or custom code is used)."""
+    def __init__(self):
+        BasePlugin.__init__(self, category="manifest")
+        self.severity = Severity.WARNING
+        self.apk_constants = None
```
Comment (dhaugh) [inline qark/plugins/manifest/min_sdk.py]: self.apk_constants is never used

File: qark/plugins/manifest/min_sdk.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        try:
+            min_sdk = apk_constants["min_sdk"]
+        except (KeyError, TypeError):
+            manifest_path = get_manifest_out_of_files(files)
+            try:
+                manifest_xml = minidom.parse(manifest_path)
+            except Exception:
+                log.exception("Failed to parse manifest file, is it valid syntax?")
+                return  # do not raise a SystemExit because other checks can still be ran
+            min_sdk = get_min_sdk(manifest_xml)
```
Comment (dhaugh) [inline qark/plugins/manifest/min_sdk.py]: Doesn't this default to 1 if it craps out?

File: tests/test_plugins/test_manifest_plugins/test_manifest_plugins.py [hunk 3]
```diff
@@ -73,3 +76,21 @@ def test_exported_tags(vulnerable_manifest_path):
...
+@pytest.mark.parametrize("apk_constants", [
+    None,
+    {"min_sdk": 8},
+])
+def test_min_sdk(apk_constants):
+    plugin = MinSDK()
+    plugin.run([os.path.join(os.path.dirname(os.path.abspath(__file__)),
+                             "test_min_sdk_tapjacking",
+                             "androidmanifest.xml")],
+               apk_constants=apk_constants)
```
Comment (dhaugh) [inline tests/test_plugins/test_manifest_plugins/test_manifest_plugins.py]: Does this test if apk_constants wasn't generated properly?

Comment (nwalsh1995) [reply tests/test_plugins/test_manifest_plugins/test_manifest_plugins.py]: It tests with apk_constants=None and apk_constants as a proper dictionary.

File: qark/plugins/manifest/min_sdk.py (hunk 2 above)
Comment (nwalsh1995) [reply qark/plugins/manifest/min_sdk.py]: It defaults to 1 if there is no min sdk defined in the manifest file.

File: qark/plugins/manifest/min_sdk.py (hunk 1 above)
Comment (nwalsh1995) [reply qark/plugins/manifest/min_sdk.py]: Removed

File: qark/plugins/manifest/min_sdk.py (hunk 2 above)
Comment (neilatli) [reply qark/plugins/manifest/min_sdk.py]: This is inline with what Android has as the default. 
https://developer.android.com/guide/topics/manifest/uses-sdk-element.html

File: tests/test_plugins/test_manifest_plugins/test_manifest_plugins.py (hunk 3 above)
Comment (neilatli) [reply tests/test_plugins/test_manifest_plugins/test_manifest_plugins.py]: You could also throw in an empty dictionary for this test. Potentially also a dict that doesn't contain a `"min_sdk"` even though there's nothing in the code currently that would use the falsey property of an empty dictionary.

**Pull request merged**